### ISSUE TITLE
Trying to restore 5s on 10.2 to 10.2

### ISSUE THREAD

**Issue by mirzank** [tihmstar/futurerestore#100]
Trying to restore a jailbroken 5s on 10.2 back to 10.2 (having some issues an erase wont fix but willing to try a fresh itunes restore). Been trying the instructions for two days and getting the same results.

Heres the log: https://pastebin.com/TAs9MG4F

Heres the command i use: ./futurerestore_macos -t iPhone6,2_2163830835908_14C92.shsh2 -b Mav7Mav8-7.21.00.Release.10.2.1.bbfw -p BuildManifest.plist -s sep-firmware.n53.RELEASE.im4p -m BuildManifest.plist iPhone_4.0_64bit_10.2_14C92_Restore.ipsw

heres what i've tried so far with no luck:
1) I have two versions of my shsh2 blobs, one for erase one for restore. generator is different in both, and i've tried adding both to nvram and using the corresponding shsh2 file with the same error.

2) Also don't believe i need nonceenabler, but still tried installing from cydia to see if i get any luck. no luck. uninstalled and installed several times with no luck.

3) Also i don't think i need the -w switch since i'm already jailbroken on 10.2, buts till tried.

4) Finally, for the bbfw file, i've tried the ones from 10.2.1, as well as 10.3 (there are 2 in 10.3, one starting with Mav7Mav8.... and the other with mav10, both with no luck.

5) Iphone basically goes into recovery, then says connect to itunes, then recovery stops and i have to use reiboot to get out of recovery.

I have also downloaded the newly release v114, but now get the following error:

Version: 7152d931c90adac8c66c1b5189231d586951d3f2 - 114
ERROR: Device is in an invalid state
[INFO] 32bit device detected
futurerestore init done
[Error] Error: failed to load shsh file iPhone6,2_2163830835908_14C92.shsh2
[Error] Fail code=-38
Failed with errorcode=-38

Again tried both erase and restore shsh2 blobs with same result. whats wierd is that the 5s is a 64bit device so not sure why its saying its a 32 bit device.

**Comment by iFred09** [comment]
Hi,

1- write the generator to the nvram
2- check it with nvram -p
3- go to recovery mode
4- launch futurerestore (with baseband Mav7Mav8 and n53 sep).

**Comment by mirzank** [comment]
Yes I've actually tried exactly that scenario multiple times with no luck. 
On checking nvram -p I see the generator written correctly. Again, I've tried both erase and update shsh's and made sure the nvram value and shsh were the same. 

I've also tried adding nvram auto-boot=false with no luck.

**Comment by mirzank** [comment]
And when you say go to recovery mode you mean setting auto boot to false and then using the reboot command in terminal I'm assuming

**Comment by mdang24zz** [comment]
i have experienced the same error with Iphone 6 plus on ios 10.3. Exact same problem

**Comment by toniqyteza** [comment]
I don't think your blobs are correct
did you save them with ticketbox?

**Comment by mdang24zz** [comment]
i saved them with tss checker

**Comment by mirzank** [comment]
I saved them with tsschecker and have verified the blobs are valid.

**Comment by mdang24zz** [comment]
ok lads, i found the solution. follow this exact steps by steps and then using the futurerestore v90.
I got the tool working. If you do not know how to get access to v90 then i will upload it

**Comment by mirzank** [comment]
I've tried to look for
Older versions but not sure where they are. Would appreciate it if you could upload.

**Comment by mdang24zz** [comment]
https://drive.google.com/open?id=0B4p3t5eJ-ccwaS1kV1h3Z3UwUXM

**Comment by mirzank** [comment]
another question - you asked if i saved my shsh with ticketbox. I was going through my downloads folder and saw ticketbox there. so now i'm wondering if i did indeed use that app! are the shsh's from that corrupt or something? I've checked mine on tsschecker and they are valid. but do you think theres a chance they could still be corrupt?

**Comment by mdang24zz** [comment]
it was another person that asked you about the ticketbox but if the tsschecker deems your shsh2 is correct then there is nothing to worry about. I also checked two of my shsh2 blobs and it is working perfectly at this moment

**Comment by mirzank** [comment]
no luck :(
getting the following error which is the same as before:

Nadirs-MBP:Downgrade nmirza$ ./futurerestore_macos -t iPhone6,2_2163830835908_14C92.shsh2 -b Mav7Mav8-7.21.00.Release.10.2.1.bbfw -p BuildManifest.plist -s sep-firmware.n53.RELEASE.im4p -m BuildManifest.plist iPhone_4.0_64bit_10.2_14C92_Restore.ipsw
Version: b35d3e1245da49950028d753b1bdb40fce5d21d5 - 90
futurerestore init done
reading ticket iPhone6,2_2163830833333_14C92.shsh2 done
[TSSC] opening BuildManifest.plist
WARNING: Unable to find BbSkeyId node
[TSSR] User specified not to request a Baseband ticket.
Request URL set to https://gs.apple.com/TSS/controller?action=2
Sending TSS request attempt 1... response successfully received
Did set sep+baseband path and firmware
[TSSC] opening BuildManifest.plist
WARNING: Unable to find BbSkeyId node
[TSSR] User specified to request only a Baseband ticket.
ERROR: Unable to get BasebandFirmware node
ERROR: Unable to find required BbGoldCertId in parameters
Request URL set to https://gs.apple.com/TSS/controller?action=2
Sending TSS request attempt 1... response successfully received
Found device in Normal mode
Entering recovery mode...
INFO: device serial number is C39LC0W2xxxx
Found device in Recovery mode
Identified device as n53ap, iPhone6,2
[Error] Devicenonce does not match APTicket nonce
[Error] maybe you forgot -w ?
Done: restoring failed.
Failed with errorcode=-20

**Comment by mirzank** [comment]
and just in case i tried the -w flag and this is what i get in a repeating loop until i force quit:

waiting for nonce: fc ca a9 7e 11 0e 6b 7b 84 4e 91 a6 50 c1 2c 52 e1 82 66 89 
Got ApNonce from device: 80 ab f7 0a 2e 68 72 24 25 a2 fe 1b 43 45 a8 bf 5c 21 7e 7f 
Got ApNonce from device: 80 ab f7 0a 2e 68 72 24 25 a2 fe 1b 43 45 a8 bf 5c 21 7e 7f 
Got ApNonce from device: 80 ab f7 0a 2e 68 72 24 25 a2 fe 1b 43 45 a8 bf 5c 21 7e 7f

**Comment by iFred09** [comment]
Ok, it's not the correct generator.

**Comment by mirzank** [comment]
Does that mean the app I used produced the wrong shsh's and are now useless? Like I mentioned I put them in tsschecker and they are valid.

**Comment by mirzank** [comment]
no? no one with an answer?

**Comment by iFred09** [comment]
You should use the same blobs which contain the same generator.

**Comment by mdang24zz** [comment]
lads, any luck in 10.2.1 or 10.3 or even 10.3.2 beta in unjaibroken state?

**Comment by mirzank** [comment]
No luck with doing a restore from 10.2 to 10.2. I imagine going from unjailbroken state on 10.2.1+ to 10.2 is an even tougher ask
